src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -415,15 +502,231 @@ private void CheckDisposedOrStarted()
             }
         }
 
-        private string GetCurlErrorString(int code)
+        private static string GetCurlErrorString(int code, bool isMulti = false)
         {
-            IntPtr ptr = Interop.libcurl.curl_easy_strerror(code);
+            IntPtr ptr = isMulti ? Interop.libcurl.curl_multi_strerror(code) : Interop.libcurl.curl_easy_strerror(code);
             return Marshal.PtrToStringAnsi(ptr);
         }
 
-        private Exception GetCurlException(int code)
+        private static Exception GetCurlException(int code, bool isMulti = false)
+        {
+            return new Exception(GetCurlErrorString(code, isMulti));
+        }
+
+        private void SetCurlCallbacks(SafeCurlHandle requestHandle, HttpRequestMessage request, IntPtr stateHandle)
+        {
+            SetCurlOption(requestHandle, CURLoption.CURLOPT_HEADERDATA, stateHandle);
+            SetCurlOption(requestHandle, CURLoption.CURLOPT_HEADERFUNCTION, (Interop.libcurl.curl_readwrite_callback)CurlReceiveHeadersCallback);
```

Comment:
This is going to have the same problem I previously mentioned: allocating a new delegate on each call, and having that delegate be collectable.